src/tlo/methods/healthsystem.py

```diff
@@ -2260,7 +2257,9 @@ def _get_events_due_today(self,) -> Tuple[List, List]:
                 # The event is not yet due (before topen)
                 hp.heappush(_list_of_events_not_due_today, next_event_tuple)
 
-                if next_event_tuple.priority == self.module.lowest_priority_considered:
+                # Only stop querying when reaching lowest_priority_considered in mode 2
+                if (self.module.mode_appt_constraints == 2) and  \
+                   (next_event_tuple.priority == self.module.lowest_priority_considered):
```

Comment:
why only stop queuing when we're in mode 2?

Reply:
This is an optimisation check which ensures we stop querying the queue as soon as we've collected all events due today for priorities which are being considered (i.e. equal or lower than the lowest_priority_considered). This however should only be done in mode 2, in which lowest_priority_considered is meaningful.